tools/crm_resource_runtime.c

```diff
@@ -1300,9 +1302,17 @@ cli_resource_restart(pcmk__output_t *out, pe_resource_t *rsc, pe_node_t *node,
         rsc = parent->parent;
     }
 
-    if (!resource_is_running_on(rsc, host)) {
-        const char *id = rsc->clone_name?rsc->clone_name:rsc->id;
-        if(host) {
+    running = resource_is_running_on(rsc, host);
+
+    if (!running) {
+        lookup_id = clone_strip(rsc->id);
+        rsc = parent->fns->find_rsc(parent, lookup_id, node, pe_find_any|pe_find_current);
+        free(lookup_id);
+        running = resource_is_running_on(rsc, host);
```

Comment:
again we have to be careful about unique clones when doing clone_strip()

also, we should probably do this only for clones. there is an obscure undocumented option to allow for groups that are only colocated, or only ordered -- meaning the group could be running on multiple nodes because it has one member on one node and another member on another node

Reply:
Fixed, on both points (though I also still need to test unique clones).